### PR TITLE
Change todo collection from array to hash

### LINKED ISSUE #4: Indexation bug
If user tries to mark todo as completed in `active` mode - nothing happens. The problem behind is that system tries to change item with index `data-index`. However `data-index` is just an index of the item in current view, not in model.

### PR DESCRIPTION
**Pull request by amogozov**
This change will help to resolve #4

### CODE REVIEW

File: lib/helpers/todosHelper.js [hunk 1]
```diff
@@ -21,67 +23,102 @@ module.exports = {
...
+	 * @param {Object} collection
 	 * @returns {boolean}
 	 */
-	areAllCompleted: function (items) {
-		return items.every(function (item) {
-			return item.isCompleted;
+	areAllCompleted: function (collection) {
+		var result = true;
+
+		Object.keys(collection).forEach(function (key) {
```
Comment (jughosta) [inline lib/helpers/todosHelper.js]: Better to use `every` to break the loop faster if  there are active items

File: lib/helpers/todosHelper.js [hunk 2]
```diff
@@ -13,6 +13,8 @@ var FILTERS = {
 
 module.exports = {
 
+	_sequence: 0,
```
Comment (jughosta) [inline lib/helpers/todosHelper.js]: Turn sequence in local variable to hide it

File: catberry_components/todo/list/template.hbs [hunk 3]
```diff
@@ -1,16 +1,16 @@
 <ul id="todo-list">
 	<!-- These are here just to show the structure of the list items -->
 	<!-- List items should get the class `editing` when editing and `completed` when marked as completed -->
-	{{#items}}
-		<li {{#if isCompleted}}class="completed"{{/if}} data-index="{{@index}}">
+	{{#each items}}
+		<li {{#if this.isCompleted}}class="completed"{{/if}} data-index="{{@key}}">
```
Comment (jughosta) [inline catberry_components/todo/list/template.hbs]: Maybe  rename `index` in `key`?

File: lib/Todo.js [hunk 4]
```diff
@@ -6,6 +6,7 @@ module.exports = Todo;
  * Creates new todo.
  * @param {string} label
  * @param {boolean} isCompleted
+ * @param {number} id
```
Comment (jughosta) [inline lib/Todo.js]: Unused

File: catberry_components/todo/list/template.hbs (hunk 3 above)
Comment (amogozov) [inline catberry_components/todo/list/template.hbs]: `@key` is a handlebars helper containing key from current `K => V` pair. So I can't get why it should be renamed.

Comment (amogozov) [inline catberry_components/todo/list/template.hbs]: oh, maybe you mean rename `data-index` to `data-key`?

File: lib/Todo.js (hunk 4 above)
Comment (amogozov) [inline lib/Todo.js]: Removed

File: lib/helpers/todosHelper.js (hunk 2 above)
Comment (amogozov) [inline lib/helpers/todosHelper.js]: Done

File: lib/helpers/todosHelper.js (hunk 1 above)
Comment (amogozov) [inline lib/helpers/todosHelper.js]: Done

File: catberry_components/todo/list/template.hbs (hunk 3 above)
Comment (amogozov) [inline catberry_components/todo/list/template.hbs]: As discussed - I have renamed all occurrences of `index` with `key`.

**Comment by amogozov** [comment]
Should be better now

**Comment by jughosta** [comment]
:+1:

**Pull request merged**